pytest_robotframework/_internal/robot/utils.py

```diff
@@ -188,25 +210,32 @@ def escape_robot_str(value: str) -> str:
     return value.replace("\\", "\\\\")
 
 
-# not using RobotOptions type since we use it to set banned options that we don't expose to the user
-def merge_robot_options(
-    obj1: Mapping[str, object], obj2: Mapping[str, object]
+def _merge_robot_options(
+    dict1: InternalRobotOptions, dict2: InternalRobotOptions
 ) -> dict[str, object]:
-    """this assumes there are no nested dicts (as far as i can tell no robot args be like that)"""
     result: dict[str, object] = {}
-    for key, value in obj1.items():
+    for key, value in dict1.items():
         if isinstance(value, list):
-            other_value = cast(Optional[List[object]], obj2.get(key, []))
-            new_value = cast(List[object], value if other_value is None else [*value, *other_value])
-        elif key in obj2:
-            new_value = obj2[key]
+            other_value = cast(Optional[List[object]], dict2.get(key, []))
+            new_value = cast(
+                Optional[List[object]],
+                other_value if other_value is None else [*value, *other_value],
+            )
+        elif key in dict2:
+            new_value = dict2[key]
         else:
             new_value = value
         result[key] = new_value
-    result.update({key: value for key, value in obj2.items() if key not in obj1})
+    result.update({key: value for key, value in dict2.items() if key not in dict1})
     return result
 
 
+def merge_robot_options(*robot_options: InternalRobotOptions) -> dict[str, object]:
+    """merges two dicts of robot options, combining lists, or overriding them with `None` if a later
+    object explicitly sets the value to `None`"""
+    return reduce(_merge_robot_options, robot_options, {})
```

Comment:
```suggestion
    return reduce(_merge_robot_options, robot_options)
```

Reply:
leaving it since there are cases where its needed